src/dial.rs

```diff
@@ -197,16 +186,19 @@ impl Dial {
         }
 
         if !self.alarm_fired && !self.in_range.contains(self.value) {
-            self.on_out_of_range();
-
-            let dial_alarm = DialAlarm::new(self.row_id, self.dial_id, Instant::now(), self.key);
+            self.alarm_fired = true;
 
-            Some(dial_alarm)
+            Some(DialAlarm::from(&*self))
         } else {
             None
         }
     }
 
+    pub fn play_alarm(&self, audio: &crate::AudioManager) {
```

Comment:
Should this actually be a separate function? We could just pass in the AudioManager as another parameter to update()

Reply:
Yeah passing audio to `Dial::update` is much cleaner